spyglass/exif.py

```diff
@@ -0,0 +1,47 @@
+def create_exif_header(orientation: int):
+    if orientation <= 0:
+        return None
+
+    return b''.join([
+        b'\xFF\xD8',  # Start of Image (SOI) marker
+        b'\xFF\xE1',  # APP1 marker
+        b'\x00\x62',  # Length of APP 1 segment (98 bytes)
+        b'\x45\x78\x69\x66',  # EXIF identifier ("Exif" in ASCII)
+        b'\x00\x00',  # Padding bytes
+        # TIFF header (with big-endian indicator)
+        b'\x4D\x4D',  # Big endian
+        b'\x00\x2A',  # TIFF magic number
+        b'\x00\x00\x00\x08',  # Offset to first IFD (8 bytes)
+        # Image File Directory (IFD)
+        b'\x00\x05',  # Number of entries in the IFD (5)
+        # v-- Orientation tag (tag number = 0x0112, type = USHORT, count = 1)
+        b'\x01\x12', b'\x00\x03', b'\x00\x00\x00\x01',
+        b'\x00', orientation.to_bytes(1, 'big'), b'\x00\x00',  # Tag data
+        # v-- XResolution tag (tag number = 0x011A, type = UNSIGNED RATIONAL, count = 1)
+        b'\x01\x1A', b'\x00\x05', b'\x00\x00\x00\x01',
+        b'\x00\x00\x00\x4A',  # Tag data (address)
+        # v-- YResolution tag (tag number = 0x011B, type = UNSIGNED RATIONAL, count = 1)
+        b'\x01\x1B', b'\x00\x05', b'\x00\x00\x00\x01',
+        b'\x00\x00\x00\x52',  # Tag data (address)
+        # v-- ResolutionUnit tag (tag number = 0x0128, type = USHORT, count = 1)
+        b'\x01\x28', b'\x00\x03', b'\x00\x00\x00\x01',
+        b'\x00\x02\x00\x00',  # 2 - Inch
+        # v-- YCbCrPositioning tag (tag number = 0x0213, type = USHORT, count = 1)
+        b'\x02\x13', b'\x00\x03', b'\x00\x00\x00\x01',
+        b'\x00\x01\x00\x00',  # center of pixel array
+        b'\x00\x00\x00\x00',  # Offset to next IFD 0
+        b'\x00\x00\x00\x48\x00\x00\x00\x01',  # XResolution value
+        b'\x00\x00\x00\x48\x00\x00\x00\x01'  # YResolution value
+    ])
+
+
+option_to_exif_orientation_map = {
```

Comment:
The correct term in python would be `dictionary` and not `map`

Reply:
I thought about it mor from a concept point of view not a technical. It maps from one to the other.

Reply:
That makes sense ofc.
Dictionary would be still a suitable term in the conceptual point of view as you are using a dictionary to understand the meaning of a word/term.
Feel free to resolve this conversation if you want to stay with map.